### ISSUE TITLE
Clarification on E1 Hard pruning: the plotted arm uses constant replacement rather than zeroing

### ISSUE THREAD

**Issue by sanghack81** [AsiaeeLab/certified-interventional-fidelity#2]
Thank you for releasing the checkpoint and experiment code. I am seeking clarification about which E1 intervention is intended by the label “Hard pruning,” because the manuscript description and the public implementation appear to differ, and the distinction materially changes the result in my rerun.

I tested the current `main` commit, `4b8359f5d0ef5fc3d2e4d1f93026b8f55160e339`, using the released MNIST checkpoint.

## Manuscript description versus implementation

[Section 4.1 (E1) of the manuscript](https://openreview.net/pdf?id=6oIS9P0Sur) describes hard pruning as “zeroing coordinates and removing the corresponding columns from the downstream weight matrix.” In the repository:

- [`plot_v2.py`](https://github.com/AsiaeeLab/certified-interventional-fidelity/blob/4b8359f5d0ef5fc3d2e4d1f93026b8f55160e339/code/plot_v2.py#L31-L36) displays `cip-const` as “Hard pruning.”
- [`pruning.py`](https://github.com/AsiaeeLab/certified-interventional-fidelity/blob/4b8359f5d0ef5fc3d2e4d1f93026b8f55160e339/code/pruning.py#L52-L67) computes a curvature-derived replacement `c_star`.
- [`make_prune_plan`](https://github.com/AsiaeeLab/certified-interventional-fidelity/blob/4b8359f5d0ef5fc3d2e4d1f93026b8f55160e339/code/pruning.py#L84-L118) returns those constants for `cip-const`.
- [`compile_const`](https://github.com/AsiaeeLab/certified-interventional-fidelity/blob/4b8359f5d0ef5fc3d2e4d1f93026b8f55160e339/code/compile.py#L14-L25) folds their downstream contribution into the bias.

Thus the plotted arm appears to be constant replacement rather than literal zeroing. In the released checkpoint, the 512 `c_star` values ranged from approximately `-31.68` to `42.43`; none used the small-curvature fallback.

## Literal-zero check

I kept the same `cip-const` selected/pruned indices and the same random trials, changing only the replacement passed to `compile_const`:

```python
implemented = compile_const(W_out, b_out, kept, pruned, constants)
literal_zero = compile_const(
    W_out, b_out, kept, pruned, torch.zeros_like(constants)
)
```

With CPU PyTorch 2.2.2, seed 42, iid sampling, and 10,000 trials, I obtained:

| `k` | mask `p` | implemented `cip-const` fidelity | literal-zero fidelity |
|---:|---:|---:|---:|
| 64 | 0.1 | 0.0985 | 0.8374 |
| 128 | 0.1 | 0.0983 | 0.9074 |
| 256 | 0.1 | 0.0994 | 0.9774 |
| 256 | 0.5 | 0.1003 | 0.5538 |

For `k=256, p=0.1`, the literal-zero version reached the paper's betting stopping conditions for `F >= 0.90` and `F >= 0.95` at trials 94 and 233, respectively. This seems different from the manuscript statement that hard pruning is omitted from the certification-cost figure because it never certifies.

Could you clarify whether “Hard pruning” is intended to mean the implemented `cip-const` constant-replacement method, or the literal-zero intervention described in the manuscript? If `cip-const` is intended, perhaps the method description and figure labels could state the replacement explicitly; if literal zeroing is intended, the reported E1 conclusion appears to change in this check.

I would be happy to share the complete diagnostic script or compare against the original execution environment, since exact stochastic paths can differ across CPU and CUDA.

_Disclosure: AI tools assisted with repository navigation and drafting. All numerical reproductions reported above were executed locally against the cited commit._
